### ISSUE TITLE
[Bug]: "Could not locate @remix-run/serve" after creating app with vercel

### ISSUE THREAD

**Issue by shpetimselaci** [remix-run/remix#676]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [X] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [X] `@remix-run/vercel`

### What version of Remix are you using?

1.0.5

### Steps to Reproduce

```
npx create-remix@latest

Where would you like to create your app? ./shp3t1ms3l4c1
Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deploym
ent targets. Vercel
TypeScript or JavaScript? TypeScript
Do you want me to run `npm install`? (Y/n) y

```
Then go ahead and try to do the linking with vercel

```
Vercel CLI 23.1.2
Set up “~/Projects/shp3t1ms3l4c1”? [Y/n] y
Which scope should contain your project? shpetimselaci
Link to existing project? [y/N] n
What’s your project’s name? shp3t1ms3l4c1
In which directory is your code located? ./
Auto-detected Project Settings (Remix):
- Build Command: remix build
- Output Directory: public
- Development Command: remix dev
Want to override the settings? [y/N] n
✅  Linked to shpetimselaci/shp3t1ms3l4c1 (created .vercel and added it to .gitignore)
➜  shp3t1ms3l4c1 vercel dev
Vercel CLI 23.1.2 dev (beta) — https://vercel.com/feedback
> Running Dev Command “remix dev”
Could not locate @remix-run/serve. Please verify you have it installed to use the dev command.

```

### Expected Behavior

It is supposed to run a server on port 3000 showing the template that remix has provided.

### Actual Behavior

I get this error when doing vercel dev or yarn dev:

```
Could not locate @remix-run/serve. Please verify you have it installed to use the dev command.
```

Which I think has to do with @remix-run/serve not being on devDependencies on the Vercel template.
Let me know whether I should make a PR for this!

**Comment by shn2016** [comment]
Never use Vercel before, but I used to get this error when I run start directly, so I assume that `vercel dev` command will need a build directory, thus you will need to `npm build`.

**Comment by girishk21** [comment]
@shpetimselaci install `@remix-run/server` https://github.com/remix-run/remix/issues/529#issuecomment-979435283, it's a known issue and Remix will push a fix probably tomorrow

**Comment by shpetimselaci** [comment]
@girishk21 thank you!
I will close this issue since there's already a PR for this.
w

**Comment by mcansh** [comment]
fixed in [1.0.6](https://github.com/remix-run/remix/releases/tag/v1.0.6)
